### PR TITLE
aiming and health bar system

### PR DESCRIPTION
**Pull request by OwenBoisse**
- Added aiming line with ray casting onto closest entity
- Added boss health bar that floats around player waist

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -75,6 +82,9 @@ pub fn main() {
         reqeusted_extensions
             .enable_fb_passthrough()
             .enable_hand_tracking();
+        reqeusted_extensions
+            .enable_fb_passthrough()
+            .enable_hand_tracking();
```
Comment (rosefromthedead) [inline src/lib.rs:87]: This and a few other hunks look to be mistaken duplications

Comment (OwenBoisse) [reply src/lib.rs:81]: should be all fixed now, had issues with merge that duplicated like half of the code

File: src/spell_control.rs
Comment (OwenBoisse) [inline src/spell_control.rs:1]: in process of changing the method in which idle spells are spawned by adding them as a child instead

Comment (OwenBoisse) [reply src/spell_control.rs:1]: all fixed in next commit

**Review by a6-webm — APPROVED** [review]
I wrote this merge, and I approve of this message

**Review by OwenBoisse — COMMENTED** [review]
merge looks good and was tested 👍

File: src/boss.rs [hunk 2]
```diff
@@ -35,13 +47,10 @@ impl Plugin for BossPlugin {
...
 }
 
-#[derive(Component)]
-struct Boss;
-
 fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
     let model = asset_server.load("white bear.glb#Scene0");
 
-    let initial_mask: u8 = DamageMask::FIRE.0 | DamageMask::LIGHTNING.0;
+    let initial_mask: DamageMask = DamageMask(DamageMask::FIRE.0 | DamageMask::LIGHTNING.0);
```
Comment (AmosHolland) [inline src/boss.rs:53]: Minor note, we haven't tested if these damage masks actually work entirely, but that's chill, just wanted it down that it's not yet been tested.

**Review by AmosHolland — APPROVED** [review]
Looks good, ideally we'd have an actual aiming system that takes into account all parts of the hand in the future, but this works for now.

**Pull request merged**